Translation loader now ignores non-directories

OverloadUT opened the pull request:
I was having an issue with habitrpg failing to load when OSX created a .DS_Store file in the locales directory. This code fixes that by ignoring all ~~hidden files (files that start with “.”)~~ non-directories.

Edit: new summary of the entire pull request:
- Locale loader will ignore anything that's not a directory (previously the entire program would crash on load if there was a file in ./locales)
- When loading individual translations, only json files will be used to import the strings

OverloadUT committed:
Translation loader now ignores system files

I was having an issue with habitrpg failing to load when OSX created a
.DS_Store file in the locales directory. This code fixes that by
ignoring all hidden files (files that start with “.”)

paglias commented:
@OverloadUT maybe just pick files with .json extension?

OverloadUT committed:
Locale loader now ignores non-directories

Locales must be stored in directories and the code previously crashed
if there were any unexpected files in the locale folder. Now the code
simply ignores all non-directories.

OverloadUT commented:
Actually, I realized that providing a list of files to ignore is silly when we could just ignore all non-directories. I just pushed a commit to do exactly that.

I also DRY'd up the code a tiny bit as the path to the locale folder was repeated in a few places.

paglias commented:
@OverloadUT fantastic! But I think it would be better to have it only load .json files, so in case there's another file (like a README, maybe) it still works. If you could add it, still keeping out directories as you did, would be fantastic

OverloadUT committed:
Only load .json files from locale directories

This opens up the possibility of putting READMEs or what have you
inside locale directories, and will avoid problems if system files or
folders end up in the locales.

OverloadUT commented:
It is done! Just committed that addition.

Summary is now:
- Locale loader will ignore anything that's not a directory (previously the entire program would crash on load if there was a file in ./locales)
- When loading individual translations, only json files will be used to import the strings

paglias commented:
@OverloadUT great, merging. UUID?

OverloadUT commented:
Here it is! d79574d4-7f75-4956-8a40-c1c863a28138

deilann commented:
Seems like you already handled the level, @paglias. :P Good thing I was conscious enough to notice.

paglias commented:
Yes :P I'll write it next time
Il 26/mag/2014 18:13 "deilann" notifications@github.com ha scritto:

> Seems like you already handled the level, @pagliashttps://github.com/paglias.
> :P Good thing I was conscious enough to notice.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/HabitRPG/habitrpg/pull/3527#issuecomment-44201577
> .
